var/spack/repos/builtin/packages/netcdf/package.py

```diff
@@ -104,91 +138,91 @@ def patch(self):
+                       '--enable-netcdf-4']
+
+        # The flag was introduced in version 4.1.0
+        if self.spec.satisfies('@4.1:'):
+            config_args.append('--enable-fsync')
+
+        # The flag was introduced in version 4.3.1
+        if self.spec.satisfies('@4.3.1:'):
+            config_args.append('--enable-dynamic-loading')
+
+        config_args += self.enable_or_disable('shared')
+
+        if '~shared' in self.spec:
             # We don't have shared libraries but we still want it to be
             # possible to use this library in shared builds
             CFLAGS.append(self.compiler.pic_flag)
 
-        if '+dap' in spec:
-            config_args.append('--enable-dap')
-        else:
-            config_args.append('--disable-dap')
-
-        if '+cdmremote' in spec:
-            config_args.append('--enable-cdmremote')
-        else:
-            config_args.append('--disable-cdmremote')
+        config_args += self.enable_or_disable('dap')
+        config_args += self.enable_or_disable('cdmremote')
 
-        if '+dap' in spec or '+cdmremote' in spec:
+        if '+dap' in self.spec or '+cdmremote' in self.spec:
             # Make sure Netcdf links against Spack's curl, otherwise it may
             # pick up system's curl, which can give link errors, e.g.:
-            #   undefined reference to `SSL_CTX_use_certificate_chain_file`
-            LIBS.append("-lcurl")
-            CPPFLAGS.append("-I%s" % spec['curl'].prefix.include)
-            LDFLAGS.append("-L%s" % spec['curl'].prefix.lib)
-
-        if '+mpi' in spec:
-            config_args.append('--enable-parallel4')
-            config_args.append('CC=%s' % spec['mpi'].mpicc)
-
-        CPPFLAGS.append("-I%s/include" % spec['hdf5'].prefix)
-        LDFLAGS.append("-L%s/lib"     % spec['hdf5'].prefix)
-
-        # HDF4 support
-        # As of NetCDF 4.1.3, "--with-hdf4=..." is no longer a valid option
-        # You must use the environment variables CPPFLAGS and LDFLAGS
-        if '+hdf4' in spec:
-            config_args.append("--enable-hdf4")
-            CPPFLAGS.append("-I%s/include" % spec['hdf'].prefix)
-            LDFLAGS.append("-L%s/lib"     % spec['hdf'].prefix)
-            LIBS.append("-l%s"         % "jpeg")
-
-        if '+szip' in spec:
-            CPPFLAGS.append("-I%s/include" % spec['szip'].prefix)
-            LDFLAGS.append("-L%s/lib"     % spec['szip'].prefix)
-            LIBS.append("-l%s"         % "sz")
-
-        # PnetCDF support
-        if '+parallel-netcdf' in spec:
-            config_args.append('--enable-pnetcdf')
-            config_args.append('CC=%s' % spec['mpi'].mpicc)
-            CPPFLAGS.append("-I%s/include" % spec['parallel-netcdf'].prefix)
-            LDFLAGS.append("-L%s/lib"      % spec['parallel-netcdf'].prefix)
+            # undefined reference to `SSL_CTX_use_certificate_chain_file
+            curl = self.spec['curl']
+            curl_libs = self.spec['curl'].libs
```

Comment:
`curl_libs = curl.libs`  (since you introduce `curl` above)

Reply:
Fixed.